### ISSUE TITLE
holdout_after does not work with value 40,000,000? works with 4,000,000

### ISSUE THREAD

**Issue by seanv507** [VowpalWabbit/vowpal_wabbit#1699]
[EDIT - markdown hid the command line params]
I am trying to use vowpalwabbit on the criteo dataset (~45 million rows - 7 days of impressions data)
what I wanted to happen is that I use first 40 million rows as training and last 5 million as holdout.
however, it does not seem to work - it claims no hold out set defined. If I use a data set of 5 million, it seems to work fine??
see log below: 
a) no hold out loss reported ( unknown)
b) average loss = undefined (no holdout)
c) number of examples per pass = 21576486, (i would have expected holdout_after - 1; for 4 million hold out_after I get 3,999,999

full data set is 45,840,617 lines.  dataset is at (https://s3.eu-central-1.amazonaws.com/test-criteo-vw/train.vw.gz)

I am using vowpalwabbit 8.6.1. as packaged in ubuntu 18.10.
`vw --final_regressor /opt/ml/model/model.bin -c --data train.vw.gz --holdout_after 40000000 --passes 30 --l2 1e-08 --link logistic --loss_function logistic --compressed`
['/opt/ml/input/data/training/train.vw.gz']
```
using l2 regularization = 1e-08
final_regressor = /opt/ml/model/model.bin
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
decay_learning_rate = 1
creating cache_file = train.vw.gz.cache
Reading datafile = train.vw.gz
num sources = 1
average since example example current current current
loss last counter weight label predict features
0.693147 0.693147 1 1.0 -1.0000 0.5000 37
0.526412 0.359676 2 2.0 -1.0000 0.3021 38
0.392392 0.258372 4 4.0 -1.0000 0.2581 25
0.525705 0.659019 8 8.0 1.0000 0.1841 29
0.512478 0.499252 16 16.0 -1.0000 0.1843 30
0.634170 0.755862 32 32.0 -1.0000 0.2725 27
0.589017 0.543863 64 64.0 1.0000 0.0978 33
0.575090 0.561164 128 128.0 -1.0000 0.3968 33
0.520033 0.464976 256 256.0 -1.0000 0.0890 29
0.521504 0.522975 512 512.0 -1.0000 0.1619 36
0.481930 0.442356 1024 1024.0 -1.0000 0.1679 33
0.475071 0.468213 2048 2048.0 -1.0000 0.3518 33
0.469413 0.463754 4096 4096.0 1.0000 0.4830 32
0.474600 0.479787 8192 8192.0 -1.0000 0.0240 33
0.472812 0.471024 16384 16384.0 -1.0000 0.6055 35
0.466881 0.460951 32768 32768.0 -1.0000 0.2717 34
0.458549 0.450216 65536 65536.0 -1.0000 0.0418 36
0.455606 0.452663 131072 131072.0 -1.0000 0.1896 37
0.471236 0.486866 262144 262144.0 -1.0000 0.4708 34
0.477819 0.484401 524288 524288.0 -1.0000 0.7471 36
0.473423 0.469028 1048576 1048576.0 -1.0000 0.5565 32
0.467879 0.462334 2097152 2097152.0 -1.0000 0.1400 32
0.465152 0.462425 4194304 4194304.0 -1.0000 0.1021 31
0.463388 0.461623 8388608 8388608.0 -1.0000 0.0832 33
0.464135 0.464883 16777216 16777216.0 -1.0000 0.1414 35
unknown unknown 33554432 33554432.0 -1.0000 0.1821 36 h
unknown unknown 67108864 67108864.0 -1.0000 0.2682 38 h
finished run
number of examples per pass = 21576486
passes used = 4
weighted example sum = 86305944.000000
weighted label sum = -42519848.000000
average loss = undefined (no holdout)
best constant = -1.079145
best constant's loss = 0.566329
total feature number = 2886790752
Training complete in 693.3070759773254 seconds.```

**Comment by JohnLangford** [comment]
This is somewhat mysterious---not sure what's going on.

**Comment by JohnLangford** [comment]
Can you provide full directions for making this reproducible?

**Comment by seanv507** [comment]
Thanks John, the vw call was hidden, updated. and shared s3 bucket with data. see above

**Comment by seanv507** [comment]
I think this is a user error! am still investigating, but looks like my vw file is cut short, so that the holdout_after is beyond the length of the file.
